client/src/client.rs

```diff
@@ -355,7 +359,7 @@ impl Client {
     ) -> Result<HashOf<VersionedTransaction>> {
         let (req, hash, resp_handler) =
             self.prepare_transaction_request::<DefaultRequestBuilder>(transaction)?;
-        let response = req
+        let response = req?
```

Comment:
It's because `req` is an implementation detail of `DefaultRequestBuilder`, not `prepare_transaction_request`. Such a pattern comes from `RequestBuilder` trait - I removed all `Result`s from it. Thus, because `DefaultRequestBuilder` may fail, it has to set its output (`req`) to `Result`.

I agree, that is not too elegant, but to solve it, I have to return `Result`s back to general `RequestBuilder` trait, that makes it less simple.

Reply:
I'm afraid I'll have to insist. The removal of `Result`s from the `RequestBuilder` is creating a hidden error handling channel. The reason why I insist on doing the extra work, is because later down the line, people might forget to handle the error.